Sources/UnionNotifications/iOS18NotificationView.swift

```diff
@@ -1,79 +1,124 @@
 
+    // MARK: Body
     var body: some View {
-        VStack(spacing: 0) {
-            VStack(spacing: 4) {
-                Text(LocalizedStringKey(localizedTitle))
-                    .font(.system(size: 17, weight: .semibold))
-                    .foregroundColor(.primary)
-                    .multilineTextAlignment(.center)
-                    .lineSpacing(2)
-                    .fixedSize(horizontal: false, vertical: true)
-
-                Text("Notifications may include alerts, sounds, and icon badges. These can be configured in Settings.", bundle: .module)
-                    .font(.system(size: 13, weight: .regular))
-                    .lineSpacing(0.4)
-                    .foregroundColor(.primary)
-                    .multilineTextAlignment(.center)
-                    .fixedSize(horizontal: false, vertical: true)
-            }
-            .padding(.vertical, 20)
-            .padding(.horizontal, 16)
-
+        GeometryReader { geometry in
             VStack(spacing: 0) {
-                Divider()
-
-                HStack(spacing: 0) {
-                    Text("Don't Allow", bundle: .module)
-                        .font(.system(size: 17, weight: .regular))
-                        .foregroundStyle(Color.blue.opacity(0.3))
-                        .frame(maxWidth: .infinity, minHeight: 44)
-
-                    Divider()
-
-                    Button {
-                        onAllow()
-                    } label: {
-                        Text("Allow", bundle: .module)
-                            .font(.system(size: 17, weight: .medium))
-                            .foregroundStyle(Color.blue)
-                            .frame(maxWidth: .infinity, minHeight: 44)
-                    }
-                    .contentShape(.rect)
-                    .buttonStyle(UnionButtonStyle(nil) { label, isPressed in
-                        label
-                            .frame(maxWidth: .infinity, minHeight: 44)
-                            .background(
-                                Rectangle()
-                                    .fill(Color.primary.opacity(isPressed ? 0.1 : 0))
-                            )
-                    })
-                }
-                .frame(height: 44)
+                // Content area
+                NotificationDialogContent(
+                    appName: appName,
+                    shouldEnableScroll: shouldEnableScroll,
+                    maxContentHeight: NotificationDialogConstants.maxScrollHeight
+                )
+                
+                // Button area
+                buttonSection
+            }
+            .background {
+                Rectangle()
+                    .fill(.regularMaterial)
             }
+            .frame(width: dialogWidth(
+                screenWidth: geometry.size.width,
+                screenHeight: geometry.size.height
+            ))
+            .clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous))
```

Comment:
This works but is redundant. You can accomplish the same thing by saying

```swift
.clipShape(.rect(cornerRadius: 16)
```

Reply:
SwiftIU rounded rectangle is continuous corner radius by default. You can check the header declaration if you want:

```swift
@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
extension Shape where Self == RoundedRectangle {

    /// A rectangular shape with rounded corners, aligned inside the frame of
    /// the view containing it.
    public static func rect(cornerSize: CGSize, style: RoundedCornerStyle = .continuous) -> Self

    /// A rectangular shape with rounded corners, aligned inside the frame of
    /// the view containing it.
    public static func rect(cornerRadius: CGFloat, style: RoundedCornerStyle = .continuous) -> Self
}
```